src/features/class-static-initializer-blocks.md

```diff
@@ -0,0 +1,108 @@
+---
+title: 'Class static initializer blocks'
+author: 'Shu-yu Guo ([@_shu](https://twitter.com/_shu))'
+avatars:
+  - 'shu-yu-guo'
+date: 2021-03-19
+tags:
+  - ECMAScript
+description: 'JavaScript classes get dedicated syntax for static initialization.'
+---
+The new class static initializer block syntax lets developers gather code that should run once for a given class definition and put them in a single place. Consider the following example where a pseudo random number generator uses a static block to initialize an entropy pool once, when the `class MyPRNG` definition is evaluated.
+
+```js
+class MyPRNG {
+  seed;
+  constructor(seed) {
+    if (seed == undefined) {
+      if (MyPRNG.entropy_pool.length === 0) {
+        throw new Error("Entropy pool exhausted");
+      }
+      seed = MyPRNG.entropy_pool.pop();
+    }
+    this.seed = seed;
+  }
```

Comment:
I'm tempted to say this should be static function `getSeed`, and then the class body gets reduced to something like:

```suggestion
  constructor(seed = MyPRNG.getSeed()) {
    this.seed = seed;
  }
```

or

```suggestion
  constructor(seed) {
    this.seed = seed ?? MyPRNG.getSeed();
  }
```

The last one seems clearer than the current approach, but maybe you think it would be too distracting?

Reply:
Sure, why not. I didn't write the example with a lot of thought. :)

Reply:
Actually I take it back. Factoring that out into a static method opens up public/private issues that I don't want to get into. There's no reason for an external user to get a seed, so `#getSeed` should be private. I don't want to be gratuitous with the number of new features being used here, and it's such a small method anyway.